sdk/php/src/DaggerConnection.php

```diff
@@ -0,0 +1,48 @@
+<?php
+
+namespace DaggerIo;
+
+use DaggerIo\Connection\CliDownloader;
+use DaggerIo\Connection\DevDaggerConnection;
+use DaggerIo\Connection\EnvSessionDaggerConnection;
+use DaggerIo\Connection\ProcessSessionDaggerConnection;
+use GraphQL\Client;
+use InvalidArgumentException;
+
+abstract class DaggerConnection
+{
+    protected static ?DevDaggerConnection $devConnectionInstance = null;
```

Comment:
Instead of hard-coding this, I think that adding support for `_EXPERIMENTAL_DAGGER_RUNNER_HOST` would be the way to go.

When there is a custom Dagger Engine, we always overwrite the default with this environment variable. This can be a dev version, or a highly specialized Dagger Engine, e.g. the one with GPU support: https://github.com/dagger/dagger/pull/5605